src/App.js

```diff
@@ -1,9 +1,19 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
+const useSemiPersistentState = (key, initialState) => {
+    const [value, setValue] = useState(JSON.parse(localStorage.getItem(key)) ?? initialState);
+    
+    useEffect(() => {
+        localStorage.setItem(key, JSON.stringify(value));
+    }, [key, value]);
+    
+    return [value, setValue];
+};
+
 const App = () => {
-    const [todoList, setTodoList] = useState([]);
+    const [todoList, setTodoList] = useSemiPersistentState("savedTodoList", []);
     const addTodo = (newTodo) => setTodoList(prevTodoList => [...prevTodoList, newTodo]);
     
     return (
```

Comment:
Nitpick (doesn't require changes):
Missing the last checklist item...
>  - [ ]  Open /src/App.js and update the JSX to use a Fragment

(change the wrapping `div` to be a fragment)

Reply:
Oh yes, I forgot about that! I'll change it. Thanks Matt!